### ISSUE TITLE
RottenTomatoes review

### ISSUE THREAD

**Issue by ngothanhtai** [ngothanhtai/RottenTomatoes#1]
Hi Charles,

This is my homework assignment.
/cc @coderschoolreview

Thanks,
Tai

**Comment by coderschoolreview** [comment]
:+1: Nice work. The point of this homework was to explore a simple example of a full MVC application with a RESTful API. You learned about UITableViews, custom UITableViewCells, navigation from a UITableView, and basic networking.
- Nice UI :wink: 
- Good clean code. Good job utilzing the best parts from the style guide:
  - [Ray Wenderlich Swift Style Guide](https://github.com/raywenderlich/swift-style-guide)
- In future homework, we will cover how to implement models that can deserialize from an NSDictionary. [SwiftyJSON](https://github.com/SwiftyJSON/SwiftyJSON) is a useful project for making it easier to deserialize the JSON response. For a more robust solution, you can also investigate using an ORM like [Pistachio](https://github.com/felixjendrusch/Pistachio).
- Nice work with the custom Movie cell and having `updateUI` func to configure itself.
- Your images are loading synchronously. This will block the UI thread and make it so that your app appears frozen while the images are being downloaded.
- Note that your images are stretched. You should observe the UIContentMode of the UIImageView. By default, it will stretch the image to match your dimensions, which is probably not what you want.
- Nice work implementing the error message on network failure. But the message looks a little odd. You can make it overlap the search bar, then hide it.
  ![12270502_10153209620407091_1284798676_n](https://cloud.githubusercontent.com/assets/15702670/11220800/387e0d56-8d94-11e5-8cb6-7bef7b414874.jpg)
- Nice job implementing a UIScrollView. UIScrollView is the parent class of UITableView.
- Nice job keeping things nice and aligned on a grid. That helps the application look cleaner and make it easier for the user to find the correct information. For a lot more information, you can read [Design tips for iOS9](https://designcode.io/iosdesign-guidelines).
- Great job with the collection view :smile: 
- For now, I cannot dismiss the keyboard when the search bar is empty. You should add `Cancel` button to handle this.
  ![12243995_10153209620417091_315480364_n](https://cloud.githubusercontent.com/assets/15702670/11220835/7db9fbfa-8d94-11e5-8002-f7cdf790464d.jpg)
- Your project organization looks a little complex. You only need to separate into 3 groups, such as: `Models`, `Views` and `View Controllers`
- Please keep your `{` and `}` correctly.

``` Swift
    static func loadPosters(thumbnailURL:String, detailedURL:String, posterImageView:UIImageView)

    {
        ImageManager.loadImage(thumbnailURL, posterImageView: posterImageView) { (thumbnailImage, fromCache) -> Void in
            posterImageView.image = thumbnailImage

           // your code
        }
    }
```
- Looks like unnecessary blank line. Don't put a blank lines after `{` and before `}`, the prefered format should be

``` Swift
someBlock {
  someOtherBlock { // no blank line here
  }
  // blank line here is okay
  someVar = someMoreCode
  ...
  // blank line here is okay
  someFinalCleanup() // no blank line at end of block either
}
```

In short, in a block, no blank line right after the start `{` and right before the `}`. Blank lines are welcome anywhere else in the body though.

Excellent submission overall :boom: Great job dedicating enough time to do the homework. You learned a lot! Keep up the good work.

**Comment by ngothanhtai** [comment]
Thank you for reviewing my code.
I really appreciate it.
